### PR TITLE
synchronize: Use SSH args from SSH connection plugins

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

SSH configuration migrated from Ansible configuration to
SSH connection configuration. Make ``synchronize`` understand
this.

Fixes: #222

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/222_synchronize.yml
plugins/action/synchronize.py
plugins/modules/synchronize.py

### CODE REVIEW

File: plugins/action/synchronize.py [hunk 1]
```diff
@@ -205,7 +213,6 @@ def run(self, tmp=None, task_vars=None):
                 "so it cannot work." % self._connection.transport)
             return result
 
-        use_ssh_args = _tmp_args.pop('use_ssh_args', None)
```
Comment (quidame) [inline plugins/action/synchronize.py:216]: this is the "too many blank lines" pep8 check complains about :)
```suggestion
```
The integration test failing on Mac OS X doesn't seem to be related to this PR.

Comment (Akasurde) [reply plugins/action/synchronize.py]: Thanks.

**Comment by Akasurde** [comment]
recheck

**Comment by Akasurde** [comment]
cc @bcoca @quidame Could you please review this? We are required to merge this before the next release cycle (which is planned for 8th August).

**Review by quidame — COMMENTED** [review]
LGTM. There's a point to clarify in the doc IMHO.

**Comment by ssbarnea** [comment]
Any updates on this? This is blocking for addressing the inability to use synchronize module with podman containers. Thanks.

**Review by bcoca — APPROVED** [review]

**Review by quidame — APPROVED** [review]

**Comment by Akasurde** [comment]
> Any updates on this? This is blocking for addressing the inability to use synchronize module with podman containers. Thanks.

I am working on CI failures. Once those are fixed. @saito-hideki will start working on release.

**Pull request merged**